### PR TITLE
abstract to methods "matplotlib-applied.ipynb"

### PR DESCRIPTION
**Pull request by Zhaochengxiann**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] [local build the book](https://github.com/open-academy/machine-learning/blob/main/CONTRIBUTING.md#build-the-book)
- [ ] review the artifacts from local build
- [ ] [local build the slides](https://github.com/open-academy/machine-learning/blob/main/CONTRIBUTING.md#build-the-slides-optional) if appliable
- [ ] Unit test if appliable

# Checklist:

- [ ] My code follows the [style guidelines](https://github.com/open-academy/machine-learning/blob/main/STYLE_GUIDE.md) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have applied tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb [hunk 1]
```diff
@@ -61,47 +61,45 @@
    "metadata": {},
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:61]: <p>Please make sure the code is executable with no error.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/173/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:61]: <pre class="ql-syntax" spellcheck="false">Line #1.    df_train = pd.read_csv('../../assets/data/titanic_train.csv')
</pre><p>Make this to be in a separate notebook cell.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/173/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:61]: <pre class="ql-syntax" spellcheck="false">Line #6.            df (_type_): _description_
</pre><p>You can fill in the <code>_type_</code> and <code>_description_</code> with the actual content.  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/173/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:61]: <pre class="ql-syntax" spellcheck="false">Line #17.        # return column_value_mapping
</pre><p>remove this if it is not used.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/173/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:61]: <pre class="ql-syntax" spellcheck="false">Line #39.        if len(df[df["Fare"].isnull()]) &gt; 0:
</pre><p>create a function as clean_fare() for line38-41</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/173/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:61]: <pre class="ql-syntax" spellcheck="false">Line #43.        # To keep Age in tact, make a copy of it called AgeFill
</pre><p>create a function as clean_age() for line43-54</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/173/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:61]: <pre class="ql-syntax" spellcheck="false">Line #1.    def survival_rate_by_Sex(gender, live, plot_title, plot_kind, plot_stacked):
</pre><p>get_survival_rate_by_gender(gender)</p><p> </p><p>And you don't need to pass into the <code>live</code>  parameter. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/173/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:61]: <pre class="ql-syntax" spellcheck="false">Line #5.        gender_xt_pct.plot(kind=plot_kind, stacked=plot_stacked, title=plot_title)
</pre><p><code>return gender_xt_pct</code> </p><p> </p><p>And move the plot code out of t he function.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/173/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:61]: done

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:61]: Teacher, I run it on my computer without reporting an error, I don’t know where the error is

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:61]: ok，Thanks for reminding

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:61]: done

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:61]: done

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:61]: done

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:61]: done

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:61]: done

File: open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb [hunk 2]
```diff
@@ -61,67 +61,78 @@
...
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:61]: <p>why some content is removed.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/173/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:61]: <p>Please always format the code.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/173/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:61]: <pre class="ql-syntax" spellcheck="false">Line #5.    #     gender_xt_pct.plot(kind=plot_kind, stacked=plot_stacked, title=plot_title)
</pre><p>please always remove the unused code.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/173/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:61]: <p>please always format the code.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/173/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:61]: <p>line 4-7 could be a function?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/173/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:61]: <pre class="ql-syntax" spellcheck="false">Line #2.    passenger_classes = np.sort(df_train["Pclass"].unique())
</pre><p>line 1 to 5 could be a function?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/173/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:61]: I removed some comments that I didn't think were necessary and some extra code that had been abstracted into methods

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:61]: ok, thanks for the reminder.

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:61]: sorry, I forgot it. Thanks a lot

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:61]: done

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:61]: done

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:61]: done

File: open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb [hunk 3]
```diff
@@ -33,6 +33,11 @@
     "* Scatter Plots, subplots\n",
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:33]: <p>You don't really need to mention the source as we have the LICENSE declare.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/173/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

**Review by zhangqi444 — APPROVED** [review]

File: open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb (hunk 3 above)
Comment (zhangqi444) [reply open-machine-learning-jupyter-book/assignments/data-science/matplotlib-applied.ipynb:33]: @Zhaochengxiann

**Pull request merged**